backend/battle/tests/test_view.py

```diff
@@ -1,9 +1,113 @@
-from django.test import TestCase
+from django.test import TestCase, Client
 from model_bakery import baker
-from battle.models import PokemonTeam
+from django.urls import reverse
+from users.models import User
+from battle.models import Battle, PokemonTeam
 from battle.battles.battle import get_winner_for
 
 
+class ListBattlesTest(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.battle = baker.make('battle.Battle')
+        self.user = User.objects.create(email='example@example.com')
+        self.user.set_password('admin')
+        self.user.save()
+
+    def test_login_user_can_acess_battle_list(self):
+        self.client.login(username=self.user.email, password='admin')
+        response = self.client.get(reverse('battle_list'))
+        self.assertEqual(response.status_code, 200)
+
+    def test_data_returns_empty_list(self):
+        self.client.login(username=self.user.email, password='admin')
+        response = self.client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(response_qs, [])
+
+    def test_data_returns_one_battle_in_list(self):
+        self.client.login(username=self.user.email, password='admin')
+        battle = baker.make('battle.Battle', creator=self.user)
+        response = self.client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(response_qs, [battle])
+
+    def test_data_returns_few_battle_ids(self):
+        self.client.login(username=self.user.email, password='admin')
+        battles = baker.make('battle.Battle', creator=self.user, _quantity=4)
+        response = self.client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(battles, response_qs)
+
+    def test_data_returns_a_lot_battle_ids(self):
+        self.client.login(username=self.user.email, password='admin')
+        battles = baker.make('battle.Battle', creator=self.user, _quantity=100)
+        response = self.client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(battles, response_qs)
+
+    def test_data_an_error_when_the_user_is_not_logged(self):
+        response = self.client.get(reverse('battle_list'))
+        self.assertEqual(response.status_code, 302)
+
+    def test_data_diff_to_verify_it_returns_exactly_len_of_battle_list(self):
+        self.client.login(username=self.user.email, password='admin')
```

Comment:
Since this is being used on almost all tests, you can move this to the `setUp`

Reply:
hmm, great! I'm going to use the auth_client you mentioned.